### PR TITLE
Drug Database Search feature built and tested (but commands yet to be linked to it) 

### PR DESCRIPTION
**Pull request by pratyushghosh**

### CODE REVIEW

File: src/main/java/seedu/address/storage/DrugSearchUtility.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    /**
+     * Setter for the FileWriter object that creates the error log file.
+     * Including a setter allows try/catch blocks to be used for exception handling.
+     */
+    private static FileWriter setF() {
+        try {
+            return new FileWriter("./src/main/resources/storage/drugSearchErrorLog.txt", true);
+        } catch (Exception e) {
+            System.out.println("Fatal error: Log file could not be created.");
+            System.exit(-1);
```
Comment (shuiherng) [inline src/main/java/seedu/address/storage/DrugSearchUtility.java]: We shouldn't use system exit in this application - attach your class to the event logger and send it there instead: you'll want to catch the File IO exception and send it as a warning to the logger instead. Avoid using pokemon exception handling as you'll end up catching alot of other things. In the event that FIle IO fails, replace it with a generic empty result.

File: src/main/java/seedu/address/storage/DrugSearchUtility.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    private static String[] tooGeneric = {"capsule","cream","emulsi","gel","hydrochloride",
+            "injection","lotion","ointment","paste","patch","pill","powder",
+            "solution","supposit","syrup","tablet"};  //keywords too generic to search for, as they will match tens of drugs
+
+    /*
+     * For writing to error log.
+     */
+
+    private static FileWriter f = setF();
+    private static PrintWriter errorLog = new PrintWriter(new BufferedWriter(f));
```
Comment (shuiherng) [inline src/main/java/seedu/address/storage/DrugSearchUtility.java]: Error log has already been provided by the program under the eventbus: attach your class to it and write there instead

File: src/main/java/seedu/address/storage/DrugSearchUtility.java (hunk 1 above)
Comment (pratyushghosh) [reply src/main/java/seedu/address/storage/DrugSearchUtility.java]: Noted. System.exit was being used just for debugging purposes, unfortunately it seems to have ended up in the final code. Will remove it.

File: src/main/java/seedu/address/storage/DrugSearchUtility.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+           String nextRecord[];
+           while ((nextRecord = csvReader.readNext()) != null) {
+               if(nextRecord[1].toLowerCase().contains(keyword.toLowerCase())) {
+                   resultsCache.add(nextRecord);
+               }
+           }
+
+           for(int i = 0; i < resultsCache.size(); i++) {
+               String[] currentRecord = resultsCache.get(i);
```
Comment (shuiherng) [inline src/main/java/seedu/address/storage/DrugSearchUtility.java]: Use a StringBuilder here for consistency with other parts of the program which perform string manipulation. In this case, we have multiple fields, so we probably want to encapsulate the result under a DrugRecord class with the fields as an enumerated variable, mapped to their values using a HashMap data structure.

File: src/main/java/seedu/address/storage/DrugSearchUtility.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+               results = results.concat("\nActive Ingredient(s): "+currentRecord[10].replace("&&",", "));
+               results = results.concat("\nClassification: "+currentRecord[4]);
+               results = results.concat("\n(for more information, enter \"moreinfo "+(i+1)+"\"");
+               results = results.concat("\n\n");
+           }
+
+           return results;
+        }
+
+        catch(Exception e) {
```
Comment (shuiherng) [inline src/main/java/seedu/address/storage/DrugSearchUtility.java]: refer to comment below at line 122

**Pull request merged**